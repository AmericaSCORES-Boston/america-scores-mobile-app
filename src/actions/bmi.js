export const START_BMI_COLLECTION = 'START_BMI_COLLECTION';
export const CONTINUE_BMI_COLLECTION = 'CONTINUE_BMI_COLLECTION';

export const startBMICollection = bmi_program_id => ({
    type: START_BMI_COLLECTION,
    bmi_program_id
});

export const continueBMICollection = bmi_continue => ({
    type: CONTINUE_BMI_COLLECTION
});
