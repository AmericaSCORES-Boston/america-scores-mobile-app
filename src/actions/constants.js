export const ADD_STUDENT = 'ADD_STUDENT';
export const ADD_STUDENT_SUCCESS = 'ADD_STUDENT_SUCCESS';
export const ADD_STUDENT_FAILURE = 'ADD_STUDENT_FAILURE';
export const FETCH_STUDENTS = 'FETCH_STUDENTS';
export const FETCH_STUDENTS_SUCCESS = 'FETCH_STUDENTS_SUCCESS';
export const FETCH_STUDENTS_FAILURE = 'FETCH_STUDENTS_FAILURE';
export const FETCH_STUDENT = 'FETCH_STUDENT';
export const FETCH_STUDENT_SUCCESS = 'FETCH_STUDENT_SUCCESS';
export const FETCH_STUDENT_FAILURE = 'FETCH_STUDENT_FAILURE';
export const SWITCH_TO_STUDENT_DETAIL_ROUTE = 'SWITCH_TO_STUDENT_DETAIL_ROUTE';
export const SWITCH_TO_STUDENTS_ROUTE = 'SWITCH_TO_DETAIL_ROUTE';
