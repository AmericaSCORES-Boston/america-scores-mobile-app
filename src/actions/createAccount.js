export const CREATE_ACCOUNT_REQUESTED = 'CREATE_ACCOUNT_REQUESTED';
export const CREATE_ACCOUNT_SUCCEEDED = 'CREATE_ACCOUNT_SUCCEEDED';
export const CREATE_ACCOUNT_FAILED = 'CREATE_ACCOUNT_FAILED';

export const createAccount = (email, username, password, first_name, last_name) => ({
  type: CREATE_ACCOUNT_REQUESTED,
  email,
  username,
  password,
  first_name,
  last_name
});

export const createAccountSuccess = () => ({
  type: CREATE_ACCOUNT_SUCCEEDED,
});

export const createAccountFailure = message => ({
  type: CREATE_ACCOUNT_FAILED,
  message
});