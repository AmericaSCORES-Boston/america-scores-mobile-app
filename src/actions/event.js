export const EVENTS_FETCH_REQUESTED = 'EVENTS_FETCH_REQUESTED';
export const EVENTS_FETCH_SUCCEEDED = 'EVENTS_FETCH_SUCCEEDED';
export const EVENTS_FETCH_FAILED = 'EVENTS_FETCH_FAILED';

export const CREATE_EVENT_REQUESTED = 'CREATE_EVENT_REQUESTED';
export const CREATE_EVENT_SUCCEEDED = 'CREATE_EVENT_SUCCEEDED';
export const CREATE_EVENT_FAILED = 'CREATE_EVENT_FAILED';

export const fetchEvents = program_id => ({
    type: EVENTS_FETCH_REQUESTED,
    program_id
});

export const fetchEventsSuccess = events => ({
    type: EVENTS_FETCH_SUCCEEDED,
    events
});

export const fetchEventsFailure = message => ({
    type: EVENTS_FETCH_FAILED,
    message
});

export const createEvent = program_id => ({
    type: CREATE_EVENT_REQUESTED,
    program_id
});

export const createEventSuccess = event => ({
    type: CREATE_EVENT_SUCCEEDED,
    event
});

export const createEventFailure = message => ({
    type: CREATE_EVENT_FAILED,
    message
});
