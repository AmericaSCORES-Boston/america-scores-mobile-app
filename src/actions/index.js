export {
  fetchSites,
  fetchSitesSuccess,
  fetchSitesFailure
} from './site';

export {
  fetchPrograms,
  fetchProgramsSuccess,
  fetchProgramsFailure,
  addProgram,
  addProgramSuccess,
  addProgramFailure
} from './program';

export {
  fetchStudents,
  fetchStudentsSuccess,
  fetchStudentsFailure
} from './student';

export {
    fetchStat,
    fetchStatSuccess,
    createStat,
    createStatSuccess,
    updateStat,
    updateStatSuccess,
    statRequestFailure
} from './studentStat';

export {
    fetchStats,
    fetchStatsSuccess,
    fetchStatsFailure
} from './stat';
