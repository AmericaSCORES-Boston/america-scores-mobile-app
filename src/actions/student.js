export const STUDENT_FETCH_REQUESTED = 'STATS_FETCH_REQUESTED';
export const STUDENT_FETCH_SUCCEEDED = 'STATS_FETCH_SUCCEEDED';
export const STUDENT_FETCH_FAILED = 'STATS_FETCH_FAILED';

export const fetchStudents = program_id => ({
  type: STUDENT_FETCH_REQUESTED,
  program_id
});

export const fetchStudentsSuccess = students => ({
  type: STUDENT_FETCH_SUCCEEDED,
  students
});

export const fetchStudentsFailure = message => ({
  type: STUDENT_FETCH_FAILED,
  message
});
