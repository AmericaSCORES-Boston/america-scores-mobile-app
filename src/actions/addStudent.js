
import { ADD_STUDENT, ADD_STUDENT_SUCCESS, ADD_STUDENT_FAILURE } from '../actions/constants';

export function addStudent(student) {
  return {
    type: ADD_STUDENT,
    student,
  };
}

export function addStudentSuccess() {
  return {
    type: ADD_STUDENT_SUCCESS,
  };
}

export function addStudentFailure() {
  return {
    type: ADD_STUDENT_FAILURE,
  };
}
