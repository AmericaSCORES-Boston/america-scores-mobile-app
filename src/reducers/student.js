import * as students from '../actions/student';

export default function studentsState(state = {}, action) {
  switch (action.type) {
    case students.STUDENT_FETCH_REQUESTED:
      return {
        ...state,
        isFetching: true
      };
    case students.STUDENT_FETCH_SUCCEEDED:
      return {
        ...state,
        isFetching: false,
        student_ids: action.students.map((student) => student.student_id),
        students: action.students
      };
    case students.STUDENT_FETCH_FAILED:
      return {
        ...state,
        isFetching: false,
        message: action.message
      };
    case students.SEARCH_STUDENT_REQUESTED:
      return {
        ...state,
        isFetching: true
      };
    case students.SEARCH_STUDENT_SUCCEEDED:
      return {
        ...state,
        isFetching: false,
        searchResults: action.students
      };
    case students.SEARCH_STUDENT_FAILED:
      return {
        ...state,
        isFetching: false,
        message: action.message
      };
    case students.CREATE_STUDENT_REQUESTED:
    case students.ADD_EXISTING_STUDENT_REQUESTED:
      return {
        ...state,
        isFetching: true
      };
    case students.CREATE_STUDENT_SUCCEEDED:
    case students.ADD_EXISTING_STUDENT_SUCCEEDED:
      return {
        ...state,
        isFetching: false,
        student_ids: [...state.student_ids, action.student.student_id],
        students: [...state.students, action.student]
      };
    case students.CREATE_STUDENT_FAILED:
    case students.ADD_EXISTING_STUDENT_FAILED:
      return {
        ...state,
        isFetching: false,
        message: action.message
      };
    default:
      return state;
  }
}