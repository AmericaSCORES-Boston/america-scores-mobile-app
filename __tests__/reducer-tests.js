import studentState from '../src/reducers/student';
import * as studentsAction from '../src/actions/student';

describe('students reducer', () => {

  let state = {
    isFetching: false,
    student_ids: {},
    students: {},
    message: null,
    searchResults: {},
  };

  //pass in the action
  state = studentState(state, studentsAction);

  it('should return the initial state', () => {
    let reducer1 = studentState(undefined, {});
    expect(reducer1).toEqual(state);
  });

  it('should handle STUDENT_FETCH_REQUESTED', () => {
    var fetchStudentsResult = studentsAction.fetchStudents(undefined);
    state = studentState(state, fetchStudentsResult);
    expect(state.isFetching).toEqual(true);
  });

  it('should handle STUDENT_FETCH_SUCCEEDED', () => {
    var student = {first_name: 'bob', last_name: 'bobber', dob: '09-12-1333'}
    state.students = [student];
    var fetchStudentsSuccessAction = studentsAction.fetchStudentsSuccess(state.students);
    //console.log(fetchStudentsSuccessAction);
    state = studentState(state, fetchStudentsSuccessAction);
    expect(state.students).toEqual([student]);
  });

  //new tests added for the sorted student list
  it('should handle CREATE_STUDENT_SUCCEEDED', () => {
    var student = {program_id: '000', first_name: 'bob', last_name: 'bobber', dob: '09-12-1993'};
    var addExistingStudentAction = studentsAction.createStudentSuccess(student);
    state = studentState(state, addExistingStudentAction);
    //console.log(state.students);
    expect(state.students.length).toEqual(2);
  });

  it('students sorted by first name after creating a student to empty list', () => {
    //clear the students list
    state.students = [];
    var student = {program_id: '000', first_name: 'Rosaline', last_name: 'A', dob: '05-05-1995'};
    var createStudentSuccessAction = studentsAction.createStudentSuccess(student);
    state = studentState(state, createStudentSuccessAction);
    //console.log(state.students);
    //new length should reflect the adding action
    expect(state.students.length).toEqual(1);
    //the student should be stored first in the array because it is the only student
    expect(state.students[0]).toEqual(student);
  });

  it('students sorted by first name after creating a student to existing list', () => {
    state.students = [{first_name: 'Pear', last_name: 'P', dob: '02-02-1999'},
    {first_name: 'Alice', last_name: 'Opal', dob: '03-03-1998'},
    {first_name: 'Bob', last_name: 'Bobber', dob: '04-04-1997'}];
    var student = {program_id: '000', first_name: 'Apple', last_name: 'A', dob: '05-05-1995'};
    var createStudentSuccessAction = studentsAction.createStudentSuccess(student);
    state = studentState(state, createStudentSuccessAction);
    console.log(state.students);
    //new length should reflect the adding action
    expect(state.students.length).toEqual(4);
    //since Apple comes after Alice, the student should be stored second in the array
    expect(state.students[1]).toEqual(student);
    //The rest of the list should still be sorted alphabetically after the new student is added
    expect(state.students[0]).toEqual({first_name: 'Alice', last_name: 'Opal', dob: '03-03-1998'});
    expect(state.students[2]).toEqual({first_name: 'Bob', last_name: 'Bobber', dob: '04-04-1997'});
    expect(state.students[3]).toEqual({first_name: 'Pear', last_name: 'P', dob: '02-02-1999'});


  });

  it('students sorted by first name more detailed comparison', () => {

    state.students = [{first_name: 'Rosaline', last_name: 'P', dob: '02-02-1999'},
    {first_name: 'Roslyn', last_name: 'Opal', dob: '03-03-1998'}];
    var student = {program_id: '000', first_name: 'Rosalind', last_name: 'A', dob: '05-05-1995'};
    var createStudentSuccessAction = studentsAction.createStudentSuccess(student);
    state = studentState(state, createStudentSuccessAction);
    //console.log(state.students);
    //new length should reflect the adding action
    expect(state.students.length).toEqual(3);
    //Test the order of the modified list
    //The rest of the list should still be sorted alphabetically after the new student is added
    expect(state.students[0]).toEqual(student);
    expect(state.students[1]).toEqual({first_name: 'Rosaline', last_name: 'P', dob: '02-02-1999'});
    expect(state.students[2]).toEqual({first_name: 'Roslyn', last_name: 'Opal', dob: '03-03-1998'});

  });

  it('student added in with same first names should not be sorted by last name', () => {

    //The sorted list should not care about last names' order
    state.students = [{first_name: 'Alice', last_name: 'P', dob: '02-02-1999'},
    {first_name: 'Alice', last_name: 'Opal', dob: '03-03-1998'},
    {first_name: 'Alice', last_name: 'Z', dob: '04-03-1998'}];
    var student = {program_id: '000', first_name: 'Alice', last_name: 'A', dob: '05-05-1995'};
    var createStudentSuccessAction = studentsAction.createStudentSuccess(student);
    state = studentState(state, createStudentSuccessAction);
    console.log(state.students);
    //new length should reflect the adding action
    expect(state.students.length).toEqual(4);
    //Test the order of the modified list
    //The rest of the list should still be sorted alphabetically after the new student is added
    expect(state.students[0]).toEqual({first_name: 'Alice', last_name: 'P', dob: '02-02-1999'});
    expect(state.students[1]).toEqual({first_name: 'Alice', last_name: 'Opal', dob: '03-03-1998'});
    expect(state.students[2]).toEqual({first_name: 'Alice', last_name: 'Z', dob: '04-03-1998'});
    expect(state.students[3]).toEqual({program_id: '000', first_name: 'Alice', last_name: 'A', dob: '05-05-1995'});

  });

  it('student name being a number should not be passed in as a number when sorting', () => {

    //Edge case: when number is passed as string, it should be sorted not as numbers
    state.students = [{first_name: '10', last_name: 'P', dob: '02-02-1999'}];
    var student = {program_id: '000', first_name: '2', last_name: 'A', dob: '05-05-1995'};
    var createStudentSuccessAction = studentsAction.createStudentSuccess(student);
    state = studentState(state, createStudentSuccessAction);
    console.log(state.students);
    //new length should reflect the adding action
    expect(state.students.length).toEqual(2);
    //Test the order of the modified list
    //10 should come first
    expect(state.students[0]).toEqual({first_name: '10', last_name: 'P', dob: '02-02-1999'});
    expect(state.students[1]).toEqual(student);
  });

});
