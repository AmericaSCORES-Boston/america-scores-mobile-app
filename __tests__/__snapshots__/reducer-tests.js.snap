exports[`students reducer should handle ADD_STUDENT 1`] = `Array []`;

exports[`students reducer should handle ADD_STUDENT_SUCCESS 1`] = `Array []`;

exports[`students reducer should handle FETCH_STUDENT 1`] = `Array []`;

exports[`students reducer should handle FETCH_STUDENT_SUCCESS 1`] = `Array []`;

exports[`students reducer should handle FETCH_STUDENTS 1`] = `Array []`;

exports[`students reducer should handle FETCH_STUDENTS_SUCCESS 1`] = `Array []`;

exports[`students reducer should return the initial state 1`] = `Object {}`;
